api/plugin/file/fileutils.go

```diff
@@ -46,7 +67,7 @@ func deleteFileAndParentDirIfNeeded(path string) {
 	dir := filepath.Dir(realPath)
 	for {
 		// Read the directory.
-		entries, err := ioutil.ReadDir(dir)
+		entries, err := os.ReadDir(dir)
 		if err != nil {
 			log.Fatal("Failed to read dir: ", err)
```

Comment:
You should probably return and error here, rather than log.Fatalf() since that bring the whole system down.